test/db.tests.js

```diff
@@ -673,6 +672,194 @@ describe('LimitDBRedis', () => {
+        })
+
+      });
+      it('should rate limit if both normal and erl rate limit are exceeded', async () => {
+        const bucketName = 'bucket_with_elevated_limits_config';
+        db.configurateBucket(bucketName, {
+          size: 1,
+          per_minute: 1,
+          elevated_limits: {
+            size: 2,
+            per_minute: 2,
+          },
+        })
+
+        // first call, still within normal rate limits
+        await takeElevatedPromise({
+          type: bucketName,
+          key: 'some_bucket_key',
+          erlIsActiveKey: 'some_erl_active_identifier'
+        }).then((result) => {
+          assert.isTrue(result.conformant);
+          assert.isFalse(result.erl_activated);
+          assert.equal(result.remaining, 0)
+        })
+        // second call, normal rate limits exceeded and erl is activated.
+        // tokens in bucket is going to be 0 after this call (size 2 - 2 calls)
+        await takeElevatedPromise({
+          type: bucketName,
+          key: 'some_bucket_key',
+          erlIsActiveKey: 'some_erl_active_identifier'
+        }).then((result) => {
+          assert.isTrue(result.conformant);
+          assert.isTrue(result.erl_activated);
+          assert.equal(result.remaining, 0);
+        })
+        // third call, erl rate limit exceeded
+        await takeElevatedPromise({
+          type: bucketName,
+          key: 'some_bucket_key',
+          erlIsActiveKey: 'some_erl_active_identifier'
+        }).then((result) => {
+          assert.isFalse(result.conformant); // being rate limited
+          assert.isTrue(result.erl_activated);
+          assert.equal(result.remaining, 0);
+        })
+      });
+      it('should deduct already used tokens from new bucket when erl is activated', async () => {
+        const bucketName = 'test-bucket';
+        await db.configurateBucket(bucketName, {
+          size: 2,
+          per_minute: 1,
+          elevated_limits: {
+            size: 10,
+            per_minute: 1,
+          }
+        });
+
+        await takeElevatedPromise({ type: bucketName, key: 'some_key ', erlIsActiveKey: 'some_erl_active_identifier' })
+        await takeElevatedPromise({ type: bucketName, key: 'some_key ', erlIsActiveKey: 'some_erl_active_identifier' })
+        await takeElevatedPromise({ type: bucketName, key: 'some_key ', erlIsActiveKey: 'some_erl_active_identifier' }).then((result) => {
+          assert.isTrue(result.conformant);
+          assert.isTrue(result.erl_activated);
+          assert.equal(result.remaining, 7); // Total used tokens so far: 3
+        });
+      })
+      it('should use default ttl if erl activation period is not configured',  async (done) => {
+        const bucketName = 'test-bucket';
+        await db.configurateBucket(bucketName, {
+          size: 2,
+          per_minute: 1,
+          elevated_limits: {
+            size: 10,
+            per_minute: 1,
+            activation_period_minutes: 15,
```

Comment:
this test should not have `activation_period_minutes` defined.

Also, we need another test checking that when it is defined, the defined ttl is applied.

Reply:
Yes - doing so made it obvious that we don't yet resolve the `activation_period_minutes` correctly